src/transformers/tokenization_prophetnet.py

```diff
@@ -0,0 +1,177 @@
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import logging
+from typing import List, Optional
+from .tokenization_bert import BertTokenizer
+from .tokenization_xlm_roberta import XLMRobertaTokenizer
+
+from .tokenization_utils import PreTrainedTokenizer
+
+
+logger = logging.getLogger(__name__)
+
+VOCAB_FILES_NAMES_EN = {"vocab_file": "vocab.txt"}
+VOCAB_FILES_NAMES_CROSS_LINGUAL = {"vocab_file": "sentencepiece.bpe.model"}
+
+PRETRAINED_VOCAB_FILES_MAP = {
+    "vocab_file": {
+        "microsoft/prophetnet-large-uncased": "https://s3.amazonaws.com/models.huggingface.co/bert/microsoft/prophetnet-large-uncased/vocab.txt",
+        "microsoft/xprophetnet-large-wiki100-cased": "https://cdn.huggingface.co/microsoft/xprophetnet-large-wiki100-cased/sentencepiece.bpe.model"
+    }
+}
+
+PRETRAINED_INIT_CONFIGURATION = {
+    "microsoft/prophetnet-large-uncased": {"do_lower_case": True, "xprophetnet_tokenizer": False},
+    "microsoft/xprophetnet-large-wiki100-cased": {"do_lower_case": False, "xprophetnet_tokenizer": True}
+}
+
+PRETRAINED_POSITIONAL_EMBEDDINGS_SIZES = {
+    "microsoft/prophetnet-large-uncased": 512,
+    "microsoft/xprophetnet-large-wiki100-cased": 512
+}
+
+
+
+class ProphetNetTokenizer(PreTrainedTokenizer):
+    r"""
+            ProphetNet inherit from BERT-tokenizer, xProphetNet  inherit from XLMR-tokenizer
+        """
+
+    vocab_files_names = VOCAB_FILES_NAMES_EN    # default english version rather than cross-lingual version.
+    pretrained_vocab_files_map = PRETRAINED_VOCAB_FILES_MAP
+    pretrained_init_configuration = PRETRAINED_INIT_CONFIGURATION
+    max_model_input_sizes = PRETRAINED_POSITIONAL_EMBEDDINGS_SIZES
+
+    def __new__(cls, **kwargs):
+        xprophetnet_tokenizer = False if 'xprophetnet_tokenizer' not in kwargs.keys() else kwargs['xprophetnet_tokenizer']
+        if xprophetnet_tokenizer:
+            super_class = XLMRobertaTokenizer
+        else:
+            super_class = BertTokenizer
+        cls = type(cls.__name__, (cls, super_class), {})
+        if xprophetnet_tokenizer:
+            cls.vocab_files_names = VOCAB_FILES_NAMES_CROSS_LINGUAL
+        else:
+            cls.vocab_files_names = VOCAB_FILES_NAMES_EN
+        return super(ProphetNetTokenizer, cls).__new__(cls)
+
+    def __init__(
+        self,
+        vocab_file,
+        xprophetnet_tokenizer=False,
+        do_lower_case=None,
+        do_basic_tokenize=None,
+        never_split=None,
+        unk_token=None,
+        sep_token=None,
+        pad_token=None,
+        cls_token=None,
+        mask_token=None,
+        tokenize_chinese_chars=None,
+        bos_token=None,
+        eos_token=None,
+        **kwargs
+    ):
+        if not xprophetnet_tokenizer:
+            # inherit from BERT tokenizer
```

Comment:
I would use the word `copied from` instead of `inherit`

Reply:
As I reviewed the codes in tokenization_bart and other files, [it](https://github.com/huggingface/transformers/blob/master/src/transformers/tokenization_bart.py#L40) seems inheriting is also acceptable. So, instead of copying every function to modify them as self._tokenizer.func, I think this old code is simpler ant avoid causing bug when copying the codes.

Reply:
Well I think both ways are okay but it's also good to hear from @LysandreJik